src/declarations/stencil-public-compiler.ts

```diff
@@ -1975,6 +1975,18 @@ export interface OutputTargetBaseNext {
   dir?: string;
 }
 
+/**
+ * This enum defines all behavioral options available for
+ * exporting elements from the `dist-custom-elements` output target.
+ */
+export enum CustomElementsExportBehavior {
+  /**
+   * All components will be re-exported from the specified directories
+   * root `index.js` barrel file.
+   */
+  BARREL = 'barrel',
```

Comment:
Went with "barrel" here since "exportAll" felt a little ambiguous. All the components get exported either way, all this behavior does is additionally export them from a single file. Not married to the name, so open to changing this if we feel there is something better

Reply:
I hadn't heard `"barrel"` before you used it the other day, is your sense that's the most used terminology for this? If there's something generally used in the rollup world or in the wider JS world I'm down for whatever is the standard.

For whatever reason it makes me think of pirates, so maybe we can go with a whole nautical theme 🛢🏴‍☠️🦜⚓️⛵️

Reply:
AFAIK, "barrel" is common term. There are even VS code extensions for generating them like "auto barrel" or "... barrel generator". But, again, I'm totally open to using another term if we have concerns with this being too general or an unfamiliar term 🤷

Reply:
FWIW, I had never heard of "barrel" either - a cursory glance at the Google makes it seem like it is indeed used in the JS/TS world, I've just managed to miss it for ~7 years 😆. I'd personally be in favor of renaming it, but I'm struggling to think of what to call it otherwise (I'd probably call it "export file" or "entrypoint" to align with https://nodejs.org/api/packages.html#package-entry-points, personally)

Reply:
@alicewriteswrongs @rwaskiewicz How do we feel about something like "single entrypoint"? That seems to align with Node's practices that Ryan called out while also giving a little more context/explicitness to the fact that this will export everything from a single file

Reply:
👍

Reply:
I think something along the lines of 'single entrypoint' makes sense...not sure precisely how to word it, maybe something like 

```ts
enum CustomElementsExportBehavior {
  DEFAULT = "default",
  SINGLE_MODULE = "single-module",
}
```

I was also wondering if in this PR we should add some code into the output target validation to ensure that the property is always set on `DIST_CUSTOM_ELEMENTS` output targets w/ the default if the user doesn't set it. Ideally we'd push that up into the type system too. But if we always knew it would be defined then I think in general it would be good to rewrite the section of the code response for actually writing files to disk to look (at a super high level) something like

```ts
switch (outputTarget.customElementsExportBehavior) {
  case (CustomElementsExportBehavior.DEFAULT):
    // do everything required for the default case
  case (CustomElementsExportBehavior.SINGLE_MODULE):
    // do everything required for this
};
```

Reply:
I was just gonna go with `SINGLE_ENTRY_POINT`, but `SINGLE_EXPORT_MODULE` could also work.

I can add some logic to the target validation to enforce this is always defined after that runs, but there might be a decent work effort to refactor the code into a switch like you're proposing since the different values we're introducing will be leveraged in different functions along the call stack. Not saying we shouldn't do it, but maybe that level of work is best broken off into its own task

Reply:
Pushed up the changes for the option rename and adding the validation logic for the custom element output target